parsers/US_IPC.py

```diff
@@ -0,0 +1,145 @@
+#!usr/bin/env python3
+
+"""Parser for the Idaho Power Comapny area of the United States."""
+
+from dateutil import parser, tz
+from itertools import groupby
+from logging import getLogger
+import requests
+
+
+# NOTE No pumped storage yet but future ideas can be found at the following url.
+# https://docs.idahopower.com/pdfs/AboutUs/PlanningForFuture/irp/IRP.pdf
+
+PRODUCTION_URL = 'https://api.idahopower.com/Energy/Api/V1/GenerationAndDemand/Subset'
+
+GENERATION_MAPPING = {'Non-Utility Geothermal': 'geothermal',
+                      'Hydro': 'hydro',
+                      'Coal': 'coal',
+                      'Diesel': 'oil',
+                      'PURPA/Non-Utility Wind': 'wind',
+                      'Natural Gas': 'gas',
+                      'PURPA/Non-Utility Solar': 'solar',
+                      'PURPA Other': 'biomass'
+                      }
+
+
+def get_data(session=None):
+    """Returns a list of dictionaries."""
+
+    s = session or requests.Session()
+
+    req = requests.get(PRODUCTION_URL)
+    json_data = req.json()
+    raw_data = json_data['list']
+
+    return raw_data
+
+
+def timestamp_converter(timestamp):
+    """Converts a str timestamp into an aware datetime object."""
+
+    dt_naive = parser.parse(timestamp)
+    mountain_time = tz.gettz("America/Boise")
+    dt_aware = dt_naive.replace(tzinfo=dt_naive.tzinfo or mountain_time)
```

Comment:
As far as I can tell, the source data never has timezone specified in its timestamps. So this check of `dt_naive.tzinfo` isn't really needed because it'll always be None. Also a new `tz.gettz` lookup is done for every timestamp, but that'll never change. We could move it to a constant up top:

```
TIMEZONE = tz.gettz("America/Boise")

....

def timestamp_converter(timestamp):
    """Converts a str timestamp into an aware datetime object."""

    dt_naive = parser.parse(timestamp)
    dt_aware = dt_naive.replace(tzinfo=TIMEZONE)

    return dt_aware
```